### PR TITLE
Add debouncing and switch to per-device coordinators

### PR DESCRIPTION
**Pull request by WebSpider**
Possibly this needs to be expanded into a debouncer per type of API call, unsure yet.

### CODE REVIEW

File: custom_components/myskoda/coordinator.py [hunk 1]
```diff
@@ -1,29 +1,51 @@
...
+        super().__init__(
+            hass,
+            _LOGGER,
+            cooldown=API_COOLDOWN_IN_SECONDS,
+            immediate=False,
+            function=func,
+        )
+
 
 class State:
```
Comment (joostlek) [inline custom_components/myskoda/coordinator.py:43]: Make this a dataclass, saves you a constrcutor

File: custom_components/myskoda/coordinator.py [hunk 2]
```diff
@@ -36,63 +58,93 @@ class MySkodaDataUpdateCoordinator(DataUpdateCoordinator[State]):
...
     data: State
-
-    def __init__(self, hass: HomeAssistant, config: ConfigEntry) -> None:
+    update_driving_range: RefreshFunction
+    update_charging: RefreshFunction
+    update_air_conditioning: RefreshFunction
+    update_vehicle: RefreshFunction
+    vin: str
+    hass: HomeAssistant
+
```
Comment (joostlek) [inline custom_components/myskoda/coordinator.py:67]: Only define the types on class level if we can't infer the types

**Review by joostlek — CHANGES_REQUESTED** [review]

File: custom_components/myskoda/coordinator.py (hunk 1 above)
Comment (Prior99) [reply custom_components/myskoda/coordinator.py:45]: Implemented in 3cac6d4

File: custom_components/myskoda/coordinator.py (hunk 2 above)
Comment (Prior99) [reply custom_components/myskoda/coordinator.py:57]: Implemented in 3cac6d4 except for `data: State` as discussed in discord

**Review by Prior99 — APPROVED** [review]

**Pull request merged**